src/Swashbuckle.AspNetCore.SwaggerGen/XmlComments/XmlCommentsTextHelper.cs

```diff
@@ -129,30 +134,68 @@ private static string HumanizeMultilineCodeTags(this string text)
 
         private static string HumanizeParaTags(this string text)
         {
-            return ParaTag().Replace(text, match =>
-            {
-                var paraText = "<br>" + match.Groups["display"].Value.Trim();
-                return LineBreaks().Replace(paraText, _ => string.Empty);
-            });
+            return ParaTag().Replace(text, match => "<br>" + match.Groups["display"].Value.Trim());
         }
 
         private static string HumanizeBrTags(this string text, string xmlCommentEndOfLine)
         {
-            return BrTag().Replace(text, _ => xmlCommentEndOfLine ?? Environment.NewLine);
+            return BrTag().Replace(text, _ => EndOfLine(xmlCommentEndOfLine));
         }
 
         private static string DecodeXml(this string text)
         {
             return WebUtility.HtmlDecode(text);
         }
 
+        private static string RemoveCommonLeadingWhitespace(string input, string xmlCommentEndOfLine)
+        {
+            var lines = input.Split(["\r\n", "\n"], StringSplitOptions.None);
+            var minLeadingSpaces = int.MaxValue;
+            foreach (var line in lines)
+            {
+                if (string.IsNullOrEmpty(line))
+                {
+                    continue;
+                }
+
+                var leadingSpaces = line.Length - line.TrimStart(' ').Length;
+                minLeadingSpaces = Math.Min(minLeadingSpaces, leadingSpaces);
```

Comment:
Is there a way to do this that avoids needing to create the intermediate strings from `TrimStart()` just to get the count? Feels you could just walk the string from the start checking the characters for being whitespace.